src/libstd/sys/unix/fd.rs

```diff
@@ -18,6 +18,11 @@ use sys::cvt;
 use sys_common::AsInner;
 use sys_common::io::read_to_end_uninitialized;
 
+#[cfg(any(target_os = "linux", target_os = "emscripten", target_os = "android"))]
+use libc::{pread64, pwrite64, off64_t};
+#[cfg(not(any(target_os = "linux", target_os = "emscripten", target_os = "android")))]
+use libc::{pread as pread64, pwrite as pwrite64, off_t as off64_t};
```

Comment:
This `off_t` as `off64_t` seems very dangerous? That'd silently cover up errors where `off_t` isn't 64-bits already.

Perhaps through it's safe to assume that the offset argument is always i64?